### PR TITLE
feat(MeshTrafficPermission): apply default deny

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Checklist prior to review

This PR changes the default behavior on no TrafficPermission and no MeshTrafficPermission for HTTP apps. It fixes a flake that we have on CI.

mtls.go "should enforce traffic permissions" (the one that already has FlakeAttempts) test is flaking in a weird way. When we apply mTLS the traffic is blocked. When we add TrafficPermission/MeshTrafficPermission it starts working, then it fails for a couple of requests and then it's running fine after this. I'd rather debug this in separate PR.

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/plugin.go [hunk 1]
```diff
@@ -74,7 +70,11 @@ func (p plugin) Apply(rs *core_xds.ResourceSet, ctx xds_context.Context, proxy *
 		}
 		rules, ok := mtp.FromRules.Rules[key]
 		if !ok {
-			continue
+			if len(proxy.Policies.TrafficPermissions) == 0 {
```
Comment (lahabana) [inline pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/plugin.go:73]: It is not supported to mix TP and MTP. Let's not add custom code trying to do their own thing in each policy

File: test/e2e_env/universal/trafficpermission/traffic_permission.go [hunk 2]
```diff
@@ -204,7 +204,7 @@ destinations:
 		Expect(err).ToNot(HaveOccurred())
 
 		// then
-		trafficBlocked()
+		trafficBlocked(503)
```
Comment (lahabana) [inline test/e2e_env/universal/trafficpermission/traffic_permission.go:207]: Would be useful to explain why this is a 503...

File: pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/plugin.go (hunk 1 above)
Comment (jakubdyszkiewicz) [reply pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/plugin.go:73]: ok, but how do you want to add a default behavior on the lack of MTP? If we miss this check, then we will override any trafficpermission

Comment (lahabana) [reply pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/plugin.go:73]: Ha you are right indeed.

**Review by lahabana — APPROVED** [review]

**Pull request merged**